modules/apps/dynamic-data-mapping/dynamic-data-mapping-form-builder/src/main/resources/META-INF/resources/metal/js/components/Calculator/Calculator.es.js

```diff
@@ -0,0 +1,231 @@
+import 'clay-dropdown';
+import {Config} from 'metal-state';
+import Component from 'metal-component';
+import Soy from 'metal-soy';
+import templates from './Calculator.soy.js';
+
+/**
+ * Calculator.
+ * @extends Component
+ */
+
+class Calculator extends Component {
+
+	static STATE = {
+		calculatorOptions: Config.arrayOf(
+			Config.shapeOf(
+				{
+					label: Config.string(),
+					tooltip: Config.string(),
+					value: Config.string()
+				}
+			)
+		).value([]),
+
+		disableCalculatorField: Config.bool().value(false),
+
+		expression: Config.string().value(''),
+
+		expressionArray: Config.array().value([]),
+
+		index: Config.number().value(0),
+
+		resultSelected: Config.string(),
+
+		options: Config.arrayOf(
+			Config.shapeOf(
+				{
+					dataType: Config.string(),
+					name: Config.string(),
+					options: Config.arrayOf(
+						Config.shapeOf(
+							{
+								label: Config.string(),
+								name: Config.string(),
+								value: Config.string()
+							}
+						)
+					),
+					type: Config.string(),
+					value: Config.string()
+				}
+			)
+		).value([]),
+
+		optionsRepeatable: Config.arrayOf(
+			Config.shapeOf(
+				{
+					dataType: Config.string(),
+					name: Config.string(),
+					options: Config.arrayOf(
+						Config.shapeOf(
+							{
+								label: Config.string(),
+								name: Config.string(),
+								value: Config.string()
+							}
+						)
+					),
+					type: Config.string(),
+					value: Config.string()
+				}
+			)
+		).value([]),
+
+		strings: {
```

Comment:
I guess it could be: 
```
strings: Config.shapeOf(
				{
					value: Config.object()
				}
			).val({
                            value: {
                               addField:  Liferay.Language.get('add-field')
                            }
                         })
```